coderd/provisionerdserver/provisionerdserver.go

```diff
@@ -271,6 +271,9 @@ func (s *server) heartbeat(ctx context.Context) error {
 }
 
 func (s *server) defaultHeartbeat(ctx context.Context) error {
+	if ctx.Err() != nil {
```

Comment:
We already do a `select` on `ctx` in `heartbeat`, don't we have some other underlying problem if we need to check err here as well? It's still as racy and it only increases the check time by a minor amount.